### ISSUE TITLE
[TODO] fix:arguments to clamp are incorrect

### ISSUE THREAD

**Issue by L-M-Sherlock** [open-spaced-repetition/fsrs-optimizer-burn#1]
https://github.com/open-spaced-repetition/fsrs-optimizer-burn/blob/4d4555ed7c00aa9a0b6e09c1c011ccf058ce3e7c/main.rs#L57

```
error[E0308]: arguments to this method are incorrect
   --> main.rs:57:31
    |
57  | ...   let new_d = new_d.clamp(1.0, 10.0); // TODO: consider constraining the associated type `<B as Backend>::FloatElem` to `{float}` or ...
    |                         ^^^^^
    |
note: expected associated type, found floating-point number
   --> main.rs:57:37
    |
57  | ...   let new_d = new_d.clamp(1.0, 10.0); // TODO: consider constraining the associated type `<B as Backend>::FloatElem` to `{float}` or ...
    |                               ^^^
    = note: expected associated type `<B as Backend>::FloatElem`
                          found type `{float}`
    = help: consider constraining the associated type `<B as Backend>::FloatElem` to `{float}` or calling a method that returns `<B as Backend>::FloatElem`
    = note: for more information, visit https://doc.rust-lang.org/book/ch19-03-advanced-traits.html
note: expected associated type, found floating-point number
   --> main.rs:57:42
    |
57  | ...   let new_d = new_d.clamp(1.0, 10.0); // TODO: consider constraining the associated type `<B as Backend>::FloatElem` to `{float}` or ...
    |                                    ^^^^
    = note: expected associated type `<B as Backend>::FloatElem`
                          found type `{float}`
    = help: consider constraining the associated type `<B as Backend>::FloatElem` to `{float}` or calling a method that returns `<B as Backend>::FloatElem`
    = note: for more information, visit https://doc.rust-lang.org/book/ch19-03-advanced-traits.html
note: method defined here
   --> /Users/jarrettye/.cargo/git/checkouts/burn-acfbee6a141c1b41/8808ee2/burn-tensor/src/tensor/api/numeric.rs:414:12
    |
414 |     pub fn clamp(self, min: K::Elem, max: K::Elem) -> Self {
    |            ^^^^^
```

**Comment by L-M-Sherlock** [comment]
@dae, @asukaminato0721, could you help me?

**Comment by dae** [comment]
This took me a while to figure out. The issue, as I understand it, is the clamp() method wants a specific type, and you're defining a generic backend type. If you limit the backend to a specific type it seems to fix the problem:

```
impl<B: Backend<FloatElem = f32>> Model<B> {
```

Unfortunately that brings Rust to its next stage of checking, and discovers a bunch of new errors regarding copying. I will try to look into this tomorrow if you don't have a chance to before me.

**Comment by dae** [comment]
This got things compiling: https://github.com/ankitects/fsrs-optimizer-burn/commits/main

**Comment by L-M-Sherlock** [comment]
Fixed by 0d67a5210147efeb206ade91716deda831d3dd10
